crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -182,6 +195,22 @@ impl TimelineEventKind {
                 }
             }
             AnySyncTimelineEvent::MessageLike(ev) => match ev.original_content() {
+                Some(AnyMessageLikeEventContent::RoomEncrypted(content)) => {
+                    // An event which is still encrypted.
+                    if let Some(unable_to_decrypt_info) = unable_to_decrypt_info {
+                        Self::UnableToDecrypt { content, unable_to_decrypt_info }
+                    } else {
+                        // If we get here, it means that some part of the code has created a
+                        // `SyncTimelineEvent` containing an `m.room.encrypted` event
+                        // without decrypting it. Possibly this means that encryption has not been
+                        // configured.
+                        // We treat it the same as any other message-like event.
+                        Self::Message {
+                            content: AnyMessageLikeEventContent::RoomEncrypted(content),
+                            relations: ev.relations(),
+                        }
+                    }
+                }
```

Comment:
Why do we handle those two cases separately? The comment doesn't really explain it to me. I'm not quite sure about why we need a new `TimelineEventKind` variant just for this…

Reply:
> I'm not quite sure about why we need a new `TimelineEventKind` variant just for this…

Well, the objective here is to convey the `unable_to_decrypt_info` that is in the `SyncTimelineEvent` or `TimelineEvent` to `TimelineEventHandler::handle_event`. It seems wrong to just shove another field onto `TimelineEventKind::Message`, hence adding a new variant.

> Why do we handle those two cases separately? The comment doesn't really explain it to me

If the lower layers have done their jobs correctly, then encrypted events will appear either as:
 * a `[Sync]TimelineEvent` with `kind: Decrypted`, in which case `ev.original_content()` here will be the decrypted event, and we hit the `Some(content) case on line 214
 * a `[Sync]TimelineEvent` with `kind: UnableToDecrypt`, in which case `ev.original_content()` here will be the *encrypted* event and `unable_to_decrypt_info` will be populated, and we hit the case at line 200.

However, the object model still allows for a `[Sync]TimelineEvent` with `kind: PlainText` whose content is actually a `AnyMessageLikeEventContent::RoomEncrypted`, and the second case (lines 203-211) is to deal with that situation. One way that I think it will happen is if the OlmMachine is not correctly set up for some reason (`BaseClient::olm_machine` is an `Option` which has to be loaded somewhere.)  And that's what the comment is trying to say.